File: scripts/mcp-refresh-probe.mjs
// SPDX-License-Identifier: Apache-2.0
import {HttpMcpClient,StdioMcpClient} from '@psp-cdl/mcpproxy/mcp';
import {McpPromptRefresher} from '@psp-cdl/llmproxy';
import {verifyEnvelope} from '@psp-cdl/core/crypto';
import {principal,binding,request} from './mcp-refresh-fixtures.mjs';
const c=JSON.parse(process.argv[2]);let peer;
try{
  peer=c.transport==='http'?await HttpMcpClient.connect(c.config,()=>c.badToken?'wrong':'synthetic-refresh-token'):await StdioMcpClient.connect(c.config);
  const client=new McpPromptRefresher(peer,{principal,sessionId:binding.sessionId,approvedCatalogDigest:c.unapproved?'wrong':peer.catalogDigest,now:()=>1100,cancelled:()=>false});
  const envelope=await client.refresh(principal,binding,request);
  verifyEnvelope(envelope,{now:1100,context:{},allowedAttributes:[],keys:[{id:'synthetic',status:'active',algorithm:'hmac-sha256',material:new Uint8Array(32).fill(19),allowUnscoped:true,scope:{},trustLevels:[2],sectionTypes:['system']}]});
  console.log(JSON.stringify({code:'OK',released:1,data:envelope.data}));
}catch(e){console.log(JSON.stringify({code:e.code??'UNEXPECTED_ERROR',released:0}));}
finally{await peer?.close();}

File: scripts/mcp_refresh_probe.py
# SPDX-License-Identifier: Apache-2.0
import json
import sys
from psp_cdl_mcpproxy.mcp import HttpMcpClient,StdioMcpClient
from psp_cdl_llmproxy import McpPromptRefresher
from psp_cdl_core.crypto import verify_envelope
from mcp_refresh_fixtures import PRINCIPAL,BINDING,REQUEST
c=json.loads(sys.argv[1]);peer=None
try:
    peer=HttpMcpClient.connect(c["config"],lambda _:"wrong" if c.get("badToken") else "synthetic-refresh-token") if c["transport"]=="http" else StdioMcpClient.connect(c["config"])
    client=McpPromptRefresher(peer,{"principal":PRINCIPAL,"sessionId":BINDING["sessionId"],"approvedCatalogDigest":"wrong" if c.get("unapproved") else peer.catalog_digest,"now":lambda:1100,"cancelled":lambda:False})
    envelope=client.refresh(PRINCIPAL,BINDING,REQUEST)
    verify_envelope(envelope,{"now":1100,"context":{},"allowedAttributes":[],"keys":[{"id":"synthetic","status":"active","algorithm":"hmac-sha256","material":bytes([19])*32,"allowUnscoped":True,"scope":{},"trustLevels":[2],"sectionTypes":["system"]}]})
    result={"code":"OK","released":1,"data":envelope["data"]}
except Exception as exc:result={"code":getattr(exc,"code","UNEXPECTED_ERROR"),"released":0}
finally:
    if peer:peer.close()
sys.stdout.buffer.write((json.dumps(result,ensure_ascii=False)+"\n").encode("utf-8"))
